plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBUtils.java

```diff
@@ -722,78 +724,56 @@ public static Object getAttributeValue(
             return null;
         }
 
+        Deque<DBDAttributeBinding> pendingAttributes = new ArrayDeque<>();
+        Deque<Integer> pendingIndices = new ArrayDeque<>();
```

Comment:
This does not introduce a substantial performance penalty but simplifies the code. Thanks to the cache, boxed integers won't be allocated if indices are in the range of 0 to 127.